src/hazelcore/CursorInfo_common.re

```diff
@@ -221,5 +259,21 @@ let get_ctx = ci => ci.ctx;
  */
 
 type deferrable('t) =
-  | CursorNotOnDeferredVarPat('t)
-  | CursorOnDeferredVarPat(UsageAnalysis.uses_list => 't, Var.t);
+  | CursorNotOnDeferred('t)
+  | CursorOnDeferredVarPat
+      // non-recursive uses
+      (
+        (
+          UsageAnalysis.uses_list,
+          // recursive uses
+          UsageAnalysis.uses_list
+        ) =>
+        't,
+        Var.t,
+      )
+  | CursorOnDeferredVarExp(
+      UsageAnalysis.uses_list => 't,
```

Comment:
Why is this parameter necessary for when the cursor is on an expression variable?